tools/src/main/scala/scala/scalanative/io/Classpath.scala

```diff
@@ -0,0 +1,55 @@
+package scala.scalanative.io
+
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.file._
+
+import scala.collection.JavaConverters._
+
+/**
+  * A values which represents a member of a classPath.
+  */
+sealed abstract class Classpath[T <: VFile] {
+  def contents: Stream[T]
+  def close(): Unit
+}
+
+/**
+  * A trait for abstracting memory cached file-like object contained in some member of a classpath.
+  */
+trait VFile {
+  def relativePath: Path
+  def bytes: ByteBuffer
+  def uri: URI
+}
+
+final case class LocalClasspath(path: Path) extends Classpath[LazyVFile[LocalClasspath]] {
+  override def close(): Unit = ()
+
+  override def contents: Stream[LazyVFile[LocalClasspath]] = Files.walk(path,Integer.MAX_VALUE, FileVisitOption.FOLLOW_LINKS)
+    .iterator()
+    .asScala.toStream
+    .map(filePath => LazyVFile[LocalClasspath](filePath, this))
```

Comment:
The very first classpath traversal will force this. We should just use a `Seq` and compute this eagerly. Scala `Stream` is very unpredictable performance-wise.